init-pants/action.yaml

```diff
@@ -1,62 +1,85 @@
 ---
-name: Initialize Pantsbuild Environment
+name: Initialize Pants
 description: |
-  Install pants and its dependencies and prepare pants caches.
+  Install Pants and its dependencies and prepare its caches.
 
 inputs:
-  # inputs are always string type
-  pants-python-version:
+  # Note: inputs are always string typed.
+  setup-commit:
+    description: |
+      Run the get-pants.sh script version at this commit in the pantsbuild/setup repo.
+      Picking a specific commit is safer than downloading the current version.
     required: false
-    default: '3.9'
+    # When we update get-pants.sh, we should update this commit.
+    default: baaf73b00d1f0c0508cfbdf4987dc9caa69d85b9
   base-branch:
+    description: |
+      The fallback commit to restore the local process cache from, if no cache
+      is available at the current commit.
     required: false
     default: main
   gha-cache-key:
-    required: true
-    # use this for matrix.python-version or similar
+    description: Qualify all cache keys with this string. Useful for invalidating everything.
+    required: false
+    default: v0
   named-caches-hash:
+    description: |
+      The hash to cache the named caches against. Should be a hash generated by hashFiles()
+      on all inputs to named caches, which are typically your lockfiles.
     required: true
-    # should be a hash generated by hashFiles()
   pants-ci-config:
+    description: |
+      An extra config file to use in CI. Defaults to pants.ci.toml if present, or no
+      file otherwise. Set to an empty string to disable.
     required: false
     default: DEFAULT
-    # defaults to pants.ci.toml if present
-    # set to empty string to disable
   cache-lmdb-store:
+    description: |
+      Whether to upload and restore the local process cache. If you're not using a fine-grained
+      remote caching service (see https://www.pantsbuild.org/docs/remote-caching), then you may
+      also want to preserve the local Pants process cache (lmdb_store). However this must
+      invalidate for changes to any file that can affect the build, so may not be practical in
+      larger repos. A remote cache service integrates with Pants's fine-grained invalidation and
+      avoids these problems.
     required: false
     default: 'false' # a string!
 
 runs:
   using: "composite"
   steps:
-    - name: 'Set up Python for Pants (${{ inputs.pants-python-version }})'
-      uses: actions/setup-python@v4
-      with:
-        python-version: '${{ inputs.pants-python-version }}'
```

Comment:
The latter. The embedded interpreter just runs Pants and rules, it is unavailable to use in Process invocations. So if you use Pants for Java and Go, you're done. If you use Pants for Python, you still need to setup Python like you always have.

Reply:
Then dropping this step removes one of the benefits of this action: hide the fact that pants uses a different version of python for itself so I can avoid pointless discussions about updating python versions.

Reply:
Is there a way to query scia-pants for the interpreter version, so that we can optionally make that same version available separately (esp for in-repo plugins)?

Reply:
This step is dropped because Pants no longer needs an interpreter. If you have Python code in your repo (whether plugin or other) you need an interpreter of course, but that's now up to you, just as installing the Go runtime or a JDK would be.

Reply:
I'm not aware of a way to query scie-pants for the underlying python interpreter, but we can document this as we don't expect it to change very often. 

It would be dangerous to automatically tie the installation of a python interpreter in CI to the interpreter version embedded in pants, since this will also affect your non-plugin code.

Reply:
I still need a way to
(A) tie my pants-plugins resolve to the python version pants is using (esp because I now have no control over that)
(B) make that python version available for testing that resolve.

I had to install a separate interpreter for pants because the st2 code supports python3.6 and pants does not. Having that in this action meant that I didn't have to worry about the pants-plugins resolve, because this action takes care of installing whatever python pants needs. Now, that assumption no longer holds. Please either let pants use the scie-pants interpreter for fmt/lint/test on a pants-plugins resolve, or make it expose a knob so that I'm testing with the same version of python that I'm pants is actually using.

Reply:
```
SCIE=inspect pants | jq .ptex
{
  "cpython-3.8.15+20221106-aarch64-apple-darwin-install_only.tar.gz": "https://github.com/indygreg/python-build-standalone/releases/download/20221106/cpython-3.8.15+20221106-aarch64-apple-darwin-install_only.tar.gz",
  "cpython-3.8.15+20221106-aarch64-unknown-linux-gnu-install_only.tar.gz": "https://github.com/indygreg/python-build-standalone/releases/download/20221106/cpython-3.8.15+20221106-aarch64-unknown-linux-gnu-install_only.tar.gz",
  "cpython-3.8.15+20221106-x86_64-apple-darwin-install_only.tar.gz": "https://github.com/indygreg/python-build-standalone/releases/download/20221106/cpython-3.8.15+20221106-x86_64-apple-darwin-install_only.tar.gz",
  "cpython-3.8.15+20221106-x86_64-unknown-linux-gnu-install_only.tar.gz": "https://github.com/indygreg/python-build-standalone/releases/download/20221106/cpython-3.8.15+20221106-x86_64-unknown-linux-gnu-install_only.tar.gz",
  "cpython-3.9.15+20221106-aarch64-apple-darwin-install_only.tar.gz": "https://github.com/indygreg/python-build-standalone/releases/download/20221106/cpython-3.9.15+20221106-aarch64-apple-darwin-install_only.tar.gz",
  "cpython-3.9.15+20221106-aarch64-unknown-linux-gnu-install_only.tar.gz": "https://github.com/indygreg/python-build-standalone/releases/download/20221106/cpython-3.9.15+20221106-aarch64-unknown-linux-gnu-install_only.tar.gz",
  "cpython-3.9.15+20221106-x86_64-apple-darwin-install_only.tar.gz": "https://github.com/indygreg/python-build-standalone/releases/download/20221106/cpython-3.9.15+20221106-x86_64-apple-darwin-install_only.tar.gz",
  "cpython-3.9.15+20221106-x86_64-unknown-linux-gnu-install_only.tar.gz": "https://github.com/indygreg/python-build-standalone/releases/download/20221106/cpython-3.9.15+20221106-x86_64-unknown-linux-gnu-install_only.tar.gz"
}
```

Reply:
https://github.com/a-scie/jump/blob/main/docs/packaging.md#environment-variables

Reply:
I think adding back an input for interpreter makes sense to cover in repo plugins like @cognifloyd has. @cognifloyd, in short it's always 3.9. it will flip to 3.11 at some Pants version in the future when Pants starts releasing scies instead of wheels to PyPI.

Reply:
Added an option to install a suitable interpreter if you ask for it explicitly.